utils.go

```diff
@@ -72,3 +72,9 @@ func logErrs(err []error) {
 		}
 	}
 }
+
+func debugLog(l string) {
```

Comment:
Hahah - no worries.  You use Println in this function, but you use Sprintf() almost every place that you call this function. Seems like it might be worth while to just make it accept a string and the arguments, then use Printf() instead of Println inside this function.